forest/cli/src/cli/attach_cmd.rs

```diff
@@ -0,0 +1,344 @@
+// Copyright 2019-2023 ChainSafe Systems
+// SPDX-License-Identifier: Apache-2.0, MIT
+
+use std::{
+    fs::{read_to_string, OpenOptions},
+    path::PathBuf,
+    str::FromStr,
+};
+
+use boa_engine::{
+    object::{FunctionBuilder, JsArray},
+    prelude::JsObject,
+    property::Attribute,
+    syntax::parser::ParseError,
+    Context, JsResult, JsValue,
+};
+use convert_case::{Case, Casing};
+use directories::BaseDirs;
+use forest_json::message::json::MessageJson;
+use forest_rpc_api::mpool_api::MpoolPushMessageResult;
+use forest_rpc_client::*;
+use fvm_shared::{address::Address, econ::TokenAmount, message::Message, METHOD_SEND};
+use num::{BigInt, Zero};
+use rustyline::{config::Config as RustyLineConfig, EditMode, Editor};
+use serde::Serialize;
+use serde_json::Value as JsonValue;
+
+use super::Config;
+
+#[derive(Debug, clap::Args)]
+pub struct AttachCommand {
+    /// Set a library directory for the Javascript scripts
+    #[arg(long)]
+    jspath: Option<PathBuf>,
+
+    /// Execute Javascript code non-interactively
+    #[arg(long)]
+    exec: Option<String>,
+}
+
+const PRELUDE_MODULE: &str = include_str!("./js/prelude.js");
+
+fn require(
+    _: &JsValue,
+    params: &[JsValue],
+    context: &mut Context,
+    jspath: &Option<PathBuf>,
+) -> JsResult<JsValue> {
+    if params.is_empty() {
+        return context.throw_error("expecting string argument");
+    }
+    let param = params.get(0).unwrap();
+
+    let module_name = param.to_string(context)?.to_string();
+    let path = if let Some(path) = jspath {
+        path.join(module_name)
+    } else {
+        PathBuf::from(module_name)
+    };
+
+    let result = if path.exists() {
+        read_to_string(path)
+    } else {
+        Ok(PRELUDE_MODULE.into())
+    };
+    match result {
+        Ok(buffer) => {
+            context.eval(&buffer).unwrap();
+
+            // Access module.exports and return as ResultValue
+            let global_obj = context.global_object().to_owned();
+            let module = global_obj.get("module", context).unwrap();
```

Comment:
Could we make those unwraps nicer? I mean, at least with an `expect` but propagating errors would be preferred if it makes sense.

Reply:
Yes, expect should be better here.

Reply:
Better use not `expect` actually and a proper error propagation instead.

Reply:
I've reworked this code and addressed an issue regarding files without a module inside (they return now `{}`).